backend/app/controllers/projects.js

```diff
@@ -163,21 +85,142 @@ module.exports = {
 
     selectOne: function (req, res, next) {
         var thunkQuery = req.thunkQuery;
+        var aggregateObject = {};
 
         co(function* () {
-            var project = yield thunkQuery(Project.select().from(Project).where(Project.id.equals(req.params.id)));
-            if (!_.first(project)) {
-                throw new HttpError(404, 'Not found');
+            var project = yield thunkQuery(Project.select().from(Project).where(Project.id.equals(req.params.id)), req.query);
+
+            if (!project) {
```

Comment:
For some strange reason this check won't work even if a project doesn't exist, because technically you get an object back even if nothing is found. So you can use this:
`if (!_.first(project))` 

I think this also is the case in `listAll()`